[Bindings] Documents direction Metadata Field (#3588)
dapr/docs: daprdocs/content/en/reference/api/bindings_api.md

> **Note:** In production never place passwords or secrets within Dapr component files. For information on securely storing and retrieving secrets using secret stores refer to [Setup Secret Store]({{< ref setup-secret-store >}})

### Binding direction (optional)

> There are scenarios where it would be useful to provide additional information to Dapr to indicate the direction supported by the Binding Component. 

Reviewer: ```suggestion
In some scenarios, it would be useful to provide additional information to Dapr to indicate the direction supported by the binding component. 
```

> This should help Dapr avoid situations where the Dapr sidecar stays in the `"wait for the app to become ready"` state waiting indefinetely for the application to become available.

Reviewer: ```suggestion
Providing the supported binding direction helps the Dapr sidecar avoid the `"wait for the app to become ready"` state, where it waits indefinitely for the application to become available.
```

> The `direction` field can be specified as part of the component's metadata. The valid values for this field are: `"input"`, `"output"`, or `"input, output"`

Reviewer: ```suggestion
You can specify the `direction` field as part of the component's metadata. The valid values for this field are: 
- `"input"`
- `"output"`
- `"input, output"`
```

Here a few scenarios when the `"direction"` metadata field could help:

> - When an application (detached from the sidecar) runs as a serverless workload and its scaled to 0, the check done by the dapr sidecar to wait for the app to become ready becomes pointless.

Reviewer: ```suggestion
- When an application (detached from the sidecar) runs as a serverless workload and is scaled to zero, the `"wait for the app to become ready"` check done by the Dapr sidecar becomes pointless.
```

> - If the detached dapr sidecar is what is scaled to 0 and the first thing the application does (before even start an http server) is reaching the sidecar the "wait for the app to become ready" gets the app and the sidecar into a dead lock of both ends waiting for each other.

Reviewer: ```suggestion
- If the detached Dapr sidecar is scaled to zero and the application reaches the sidecar (before even starting an HTTP server), the `"wait for the app to become ready"` deadlocks the app and the sidecar into waiting for each other.
```
